src/openms/source/FORMAT/MascotGenericFile.cpp

```diff
@@ -337,34 +337,66 @@ namespace OpenMS
       os << "BEGIN IONS\n";
       if (!store_compact_)
       {
-        os << "TITLE=" << precisionWrapper(mz) << "_" << precisionWrapper(rt)
-           << "_" << spec.getNativeID() << "_" << filename << "\n";
+        // if a TITLE is available, it was (most likely) parsed from an MGF file
+        // in that case, a nativeID was constructed from the TITLE and the spectrum index
+        if (spec.metaValueExists("TITLE"))
+        {
+          if (String(spec.getMetaValue("TITLE")).hasSubstring("_index="))
+          {
+            os << "TITLE=" << spec.getMetaValue("TITLE") << "\n";
+          }
+          else
+          {
+            os << "TITLE=" << spec.getMetaValue("TITLE") << "_" << spec.getNativeID() << "\n";
+          }
+        }
+        else
+        {
+          os << "TITLE=" << precisionWrapper(mz) << "_" << precisionWrapper(rt)
+             << "_" << spec.getNativeID() << "_" << filename << "\n";
+        }
         os << "PEPMASS=" << precisionWrapper(mz) <<  "\n";
         os << "RTINSECONDS=" << precisionWrapper(rt) << "\n";
-  if (native_id_type_accession == "UNKNOWN")
-  {
-    os << "SCANS=" << spec.getNativeID().substr(spec.getNativeID().find_last_of("=")+1) << "\n";
-  }
-  else
-  {
-          os << "SCANS=" << SpectrumLookup::extractScanNumber(spec.getNativeID(), native_id_type_accession) << "\n";
-  }
+        if (native_id_type_accession == "UNKNOWN")
+        {
+          os << "SCANS=" << spec.getNativeID().substr(spec.getNativeID().find_last_of("=")+1) << "\n";
+        }
+        else
+        {
+                os << "SCANS=" << SpectrumLookup::extractScanNumber(spec.getNativeID(), native_id_type_accession) << "\n";
+        }
       }
       else
       {
-        os << "TITLE=" << fixed << setprecision(HIGH_PRECISION) << mz << "_"
-           << setprecision(LOW_PRECISION) << rt << "_"
-           << spec.getNativeID() << "_" << filename << "\n";
+        // if a TITLE is available, it was (most likely) parsed from an MGF file
+        // in that case, a nativeID was constructed from the TITLE and the spectrum index
+        if (spec.metaValueExists("TITLE"))
```

Comment:
This code duplication is quite ugly.
The `store_compact_` flag somewhat provokes this, but can you just wrap the behavior of this flag into an inline function?! would make the code much more maintainable

Reply:
@enetz Can you "quickly" do that if you agree?

Reply:
@jpfeuffer the code duplication that @cbielow mentioned was already reduced by a previous commit.
The remaining duplicated code here is just this:
       ```
 if (spec.metaValueExists("TITLE"))
        {
          os << "TITLE=" << spec.getMetaValue("TITLE") << "\n";
        }
        else
```
and the `else` cases are different